src/Arbitrary.hs

```diff
@@ -216,39 +219,40 @@ instance Arbitrary ViewIdea where
     shrink (ViewIdea ctx ideaList) = ViewIdea <$> shr ctx <*> shr ideaList
 
 instance Arbitrary CreateIdea where
-    arbitrary = CreateIdea <$> arb
-    shrink (CreateIdea x) = CreateIdea <$> shr x
+    arbitrary = garbitrary
+    shrink    = gshrink
 
 instance Arbitrary EditIdea where
-    arbitrary = EditIdea <$> arb
-    shrink (EditIdea x) = EditIdea <$> shr x
+    arbitrary = garbitrary
+    shrink    = gshrink
 
 instance Arbitrary Frontend.Page.MoveIdea where
-    arbitrary = MoveIdea <$> arb <*> (getNonEmpty <$> arb)
-    shrink (MoveIdea x y) = MoveIdea <$> shr x <*> shr y
+    arbitrary = MoveIdea <$> arb <*> arb <*> (getNonEmpty <$> arb)
+    shrink (MoveIdea x y z) = MoveIdea <$> shr x <*> shr y <*> shr z
 
 instance Arbitrary ReportIdea where
-    arbitrary = ReportIdea <$> arb
-    shrink (ReportIdea x) = ReportIdea <$> shr x
+    arbitrary = garbitrary
+    shrink    = gshrink
 
 instance Arbitrary CommentOnIdea where
-    arbitrary = CommentOnIdea <$> arb <*> arb
-    shrink (CommentOnIdea x y) = CommentOnIdea <$> shr x <*> shr y
+    arbitrary = garbitrary
+    shrink    = gshrink
 
 instance Arbitrary EditComment where
-    arbitrary = EditComment <$> arb <*> arb
+    arbitrary = garbitrary
+    shrink    = gshrink
 
 instance Arbitrary JudgeIdea where
-    arbitrary = JudgeIdea <$> arb <*> arb <*> arb
-    shrink (JudgeIdea x y z) = JudgeIdea <$> shr x <*> shr y <*> shr z
+    arbitrary = garbitrary
+    shrink    = gshrink
 
 instance Arbitrary CreatorStatement where
-    arbitrary = CreatorStatement <$> arb
-    shrink (CreatorStatement x) = CreatorStatement <$> shr x
+    arbitrary = garbitrary
+    shrink    = gshrink
 
 instance Arbitrary ReportComment where
-    arbitrary = ReportComment <$> arb
-    shrink (ReportComment x) = ReportComment <$> shr x
+    arbitrary = garbitrary
+    shrink    = gshrink
```

Comment:
this is nicer, but it may reduce test coverage, because we are not very confident with the `garbitrary` implementation (see issue over at sop-basic).  i would prefer to keep the ugly cases here because they are less likely to cause follow-up work.